packages/compass-components/src/components/bson-document-editor.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useRef, useState } from 'react';
+import { css } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import type { EditorProps } from './editor';
+import { Editor } from './editor';
+import queryParser from 'mongodb-query-parser';
+import { EJSON } from 'bson';
+import { useElementParentHoverState } from '../utils/use-element-parent-hover-state';
+import { Button } from './leafygreen';
+
+type BSONValue = any;
+
+type BSONDocumentEditorProps = Omit<EditorProps, 'variant'> & {
+  variant?: never;
+  onChangeValue?: (
+    value: BSONValue,
+    error: Error | null,
+    variant: 'EJSON' | 'Shell',
+    text: string,
+    event?: any
+  ) => void;
+} & (
+    | {
+        initialValue?: never;
+      }
+    | {
+        text?: never;
+        initialValue?: BSONValue;
+      }
+  );
+
+const actionsGroupContainer = css({
+  position: 'absolute',
+  display: 'flex',
+  gap: spacing[2],
+  width: '100%',
+  top: spacing[3],
+  right: spacing[3],
+  paddingLeft: spacing[3],
+  paddingRight: spacing[3],
+  pointerEvents: 'none',
+  zIndex: 100,
+});
+
+const actionsGroupItem = css({
+  flex: 'none',
+  pointerEvents: 'all',
+});
+
+function BSONDocumentEditor({
+  initialValue,
+  onChangeValue,
+  text,
+  onChangeText,
```

Comment:
Do we use this anywhere (or can we see ourselves using it)? I am wondering if we can simplify the interface by only having one `onChange` here instead of two separate ones